### PR TITLE
Deployment models for Tyk Self-managed

### PR DESCRIPTION
**Pull request by SonjaChevre**
Bringing the content from https://tyktech.atlassian.net/wiki/spaces/EN/pages/1896841234/Deployment+models+for+Tyk+Self-managed to the external documentation

* added three diagrams for self-managed deployment
* updated the text on the licensing page for self-managed and renamed the title to "Licensing and deployment models"
* improved the licensing snippet for self-managed (from "free" to "free trial")

see also https://tyktech.atlassian.net/browse/DX-158

### Review link 
https://deploy-preview-2173--tyk-docs.netlify.app/docs/nightly/tyk-on-premises/licensing/

### CODE REVIEW

File: tyk-docs/content/tyk-on-premises/licensing.md [hunk 1]
```diff
@@ -14,4 +14,60 @@ aliases:
...
+
+Are you still figuring out your requirements for your deployment around performance, resilience, availability, latency, or multi-region deployment? Look at our [API management architectural and deployment patterns 101](https://tyk.io/blog/res-api-management-architectural-and-deployment-patterns-101/) post to learn more. 
+
+## 3 questions to ask yourself to select your deployment model
+
+### 1. How many environments do you manage?
+
+An environment is a single grouping of a control plane and multiple gateways, representing your development lifecycle. For example, staging and production would be 2 environments. 
+
+If you have teams working on two different projects, deployed in different environments, you might have 4 different environments (staging and prod for project number 1, staging and prod for project number 2).
```
Comment (sedkis) [inline tyk-docs/content/tyk-on-premises/licensing.md]: Or, you can take advantage of RBAC to onboard multiple teams onto the same environment as that is more price effective.  You can read more about that [here](link to a blog coming out this week about environments and RBAC)

File: tyk-docs/content/tyk-on-premises/licensing.md [hunk 2]
```diff
@@ -14,4 +14,60 @@ aliases:
...
+
+You get extra developer licenses (a single control plane with 1 connected gateway for local testing on a developer's workstation) for development and testing on local computers. 
+
+### 2. How many gateways do you need per environment?
+
+We recommend a minimum of 2 gateways per environment. Rolling updates or blue green deployment might require additional gateways to ensure high availability during deployments.
```
Comment (sedkis) [inline tyk-docs/content/tyk-on-premises/licensing.md]: "per environment for redundancy & fail over"

**Review by letzya — CHANGES_REQUESTED** [review]
Kindly please assign a peer to review and approve. Thanks

**Comment by letzya** [comment]
I personally think that "Need more" is not clear enough that we one can have many regions, I would clearly state it.

**Review by letzya — COMMENTED** [review]
another comment is that it is not clear why the package is relevant for multiple teams. MAybe we should add links to the relevant docs? (link this one for mdcb https://tyk.io/docs/tyk-multi-data-centre/#organisational-benefits)

>

**Comment by SonjaChevre** [comment]
@letzya I have addressed your change requests, please check

**Comment by SonjaChevre** [comment]
@sedkis I have also added you officially as reviewer, because I couldn't find your comments :D

**Comment by letzya** [comment]
@SonjaChevre do we want a link to the pricing page? https://tyk.io/price-comparison/

**Review by letzya — APPROVED** [review]
Suggest link to pricing

File: tyk-docs/content/tyk-on-premises/licensing.md [hunk 3]
```diff
@@ -14,4 +14,58 @@ aliases:
   - getting-started/licensing
 ---
 
-{{< include "self-managed-licensing-include" >}}
+
+Tyk Self-Managed is the easiest way to install Tyk Full Lifecycle API Management solution in your infrastructure. There is no calling home, and there are no usage limits. You have full control. 
+
+When selecting a [Tyk Self-managed license](https://tyk.io/price-comparison), you can choose between three deployment models: [Launch]({{< ref "licensing.md#launch---perfect-to-get-started" >}}), [Flex]({{< ref "licensing.md#flex---for-multiple-teams-within-a-single-data-centre-or-cloud" >}}) and [Global Multi-team]({{< ref "licensing.md#global-multi-team---for-the-largest-global-enterprises-at-scale" >}}). Read below to learn which model fits best for your scenario.
```
Comment (letzya) [inline tyk-docs/content/tyk-on-premises/licensing.md]: is this the link `licensing.md#flex---for-multiple-teams-within-a-single-data-centre-or-cloud`? in the preview I see https://tyk.io/docs/nightly/tyk-on-premises/licensing/?__hstc=76629258.0441fb021a60c922d72bfb9681eb1faf.1673279869058.1673531891531.1673534113789.4&__hssc=76629258.2.1673534113789&__hsfp=3974607804#flex---for-multiple-teams-within-a-single-data-centre-or-cloud

Comment (letzya) [reply tyk-docs/content/tyk-on-premises/licensing.md:20]: also I dont get diff link to flex and launch

Comment (letzya) [reply tyk-docs/content/tyk-on-premises/licensing.md]: also I dont get diff link to flex and launch

Comment (SonjaChevre) [reply tyk-docs/content/tyk-on-premises/licensing.md]: no, that was the link for "Tyk Self-managed license" - but that wasn't very clear. 
Changing it to "(see the pricing page to compare the licenses)"

Comment (SonjaChevre) [reply tyk-docs/content/tyk-on-premises/licensing.md]: `licensing.md#flex---for-multiple-teams-within-a-single-data-centre-or-cloud` - those are links to the same page, to quickly jump to the different deployment model. on my local environment the links work correctly, but not on the nightly build.

**Review by letzya — CHANGES_REQUESTED** [review]

**Review by letzya — APPROVED** [review]

**Pull request merged**